test/sync_tests/conflicting.test.js

```diff
@@ -0,0 +1,763 @@
+        mutator.updated_at = Factory.yesterday();
+      },
+      undefined,
+      undefined,
+      syncOptions
+    );
+    this.expectedItemCount++;
+  });
+
+  it('if server says deleted but client says not deleted, keep server state', async function () {
+    const note = await Factory.createMappedNote(this.application);
+    const originalPayload = note.payloadRepresentation();
+    this.expectedItemCount++;
+    await this.application.itemManager.setItemDirty(note.uuid);
+    await this.application.syncService.sync(syncOptions);
+    expect(this.application.itemManager.items.length).to.equal(
+      this.expectedItemCount
+    );
+
+    // client A
+    await this.application.itemManager.setItemToBeDeleted(note.uuid);
+    await this.application.syncService.sync(syncOptions);
+    this.expectedItemCount--;
+    expect(this.application.itemManager.items.length).to.equal(
+      this.expectedItemCount
+    );
+
+    // client B
+    await this.application.syncService.clearSyncPositionTokens();
+    // Add the item back and say it's not deleted
+    const mutatedPayload = CreateMaxPayloadFromAnyObject(originalPayload, {
+      deleted: false,
+      updated_at: Factory.yesterday(),
+    });
+    await this.application.itemManager.emitItemsFromPayloads(
+      [mutatedPayload],
+      PayloadSource.LocalChanged
+    );
+    const resultNote = this.application.itemManager.findItem(note.uuid);
+    expect(resultNote.uuid).to.equal(note.uuid);
+    await this.application.itemManager.setItemDirty(resultNote.uuid);
+    await this.application.syncService.sync(syncOptions);
+
+    // We expect that this item is now gone for good, and a duplicate has not been created.
+    expect(this.application.itemManager.items.length).to.equal(
+      this.expectedItemCount
+    );
+  });
+
+  it('if server says not deleted but client says deleted, keep server state', async function () {
+    const note = await Factory.createMappedNote(this.application);
+    await this.application.itemManager.setItemDirty(note.uuid);
+    this.expectedItemCount++;
+
+    // client A
+    await this.application.syncService.sync(syncOptions);
+    expect(this.application.itemManager.items.length).to.equal(
+      this.expectedItemCount
+    );
+
+    // client B
+    await this.application.syncService.clearSyncPositionTokens();
+
+    // This client says this item is deleted, but the server is saying its not deleted.
+    // In this case, we want to keep the server copy.
+    await this.application.changeAndSaveItem(
+      note.uuid,
+      (mutator) => {
+        mutator.setDeleted();
+        mutator.updated_at = Factory.yesterday();
+      },
+      undefined,
+      undefined,
+      syncOptions
+    );
+
+    // We expect that this item maintained.
+    expect(this.application.itemManager.items.length).to.equal(
+      this.expectedItemCount
+    );
+  });
+
+  it('should create conflict if syncing an item that is stale', async function () {
```

Comment:
What constitutes a stale item?

Reply:
It's mostly this: `mutator.updated_at = Factory.yesterday()`